src/coreclr/jit/emitarm64.cpp

```diff
@@ -4252,9 +4252,15 @@ void emitter::emitIns_Mov(
         {
             if (isPredicateRegister(dstReg) && isPredicateRegister(srcReg))
             {
-                assert(insOptsNone(opt));
+                if (insOptsNone(opt))
```

Comment:
This code is odd. This is because for predicates we always use INS_OPTS_SCALABLE_B for a mov, so the passed in `opt` value doesn't matter ?

Reply:
Yes, so basically, this code operates just on `INS_OPTS_SCALABLE_B` (and we added single test for that). For other places, if we have to pass `INS_OPTS_SCALABLE_B` if register types are predicate feels like an overhead, so instead this code expects it to be NONE, in which case it will set the options to `INS_OPTS_SCALABLE_B`.

Reply:
Would something like this work?
```
if (!insOptsNone(opt))
{
    assert(opt == INS_OPTS_SCALABLE_B);
}

opt = INS_OPTS_SCALABLE_B;
```

I think that shape makes it clearer the if-statement is debug-only code, and ensures `opt` is always `INS_OPTS_SCALABLE_B` even if we don't have asserts on to catch improper handling.

Reply:
```
assert(opt == INS_OPTS_SCALABLE_B || insOptsNone(opt));
opt = INS_OPTS_SCALABLE_B;
```
?